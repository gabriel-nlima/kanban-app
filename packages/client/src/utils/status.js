export const TODO = 'TODO'
export const BEING_DONE = 'BEING_DONE'
export const FINISHED = 'FINISHED'
export const FILED = 'FILED'
export const DELETED = 'DELETED'
export const actions = [TODO, BEING_DONE, FINISHED, FILED, DELETED]
