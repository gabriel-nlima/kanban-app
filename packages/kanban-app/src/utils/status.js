export const FAZER = 'A_FAZER'
export const FAZENDO = 'FAZENDO'
export const CONCLUIDO = 'CONCLUIDO'
export const ARQUIVADO = 'ARQUIVADO'
export const DELETADO = 'DELETADO'
export const acoes = [FAZER, FAZENDO, CONCLUIDO, ARQUIVADO, DELETADO]
