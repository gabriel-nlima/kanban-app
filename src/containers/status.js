export const FAZER = 'A_FAZER'
export const FAZENDO = 'FAZENDO'
export const CONCLUIDO = 'CONCLUIDO'
export const ARQUIVADO = 'ARQUIVADO'
