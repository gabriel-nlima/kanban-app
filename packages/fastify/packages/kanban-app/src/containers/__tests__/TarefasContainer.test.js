import React from 'react'
import { cleanup, fireEvent } from 'react-testing-library'

import TarefasContainerConnected, {
	TarefasContainer,
} from '../TarefasContainer'
import * as status from '../status'

import renderWithRedux, { renderWithRouter } from '../../utils/utils'

describe('Tarefas', () => {
	afterEach(cleanup)

	test('renderiza o quadro de tarefas', () => {
		//Arrange
		//Act
		const { getByText } = renderWithRedux(<TarefasContainerConnected />)

		//Assert
		const aFazer = getByText('A FAZER')
		const fazendo = getByText('FAZENDO')
		const feito = getByText('FEITO')
		const arquivadas = getByText('ARQUIVADAS:')

		expect(aFazer).toBeDefined()
		expect(fazendo).toBeDefined()
		expect(feito).toBeDefined()
		expect(arquivadas).toBeDefined()
	})
	test('deve renderizar um quadro de tarefas com 3 tarefas', () => {
		const mockTrocaStatus = jest.fn()
		const mockGet = jest.fn()

		const tarefas = [
			{
				titulo: 'fazer algo1',
				conteudo: 'fazer alguma coisa1',
				status: status.FAZER,
				adicionadoEm: '28/01',
			},
			{
				titulo: 'fazer algo2',
				conteudo: 'fazer alguma coisa2',
				status: status.FAZENDO,
				adicionadoEm: '28/01',
			},
			{
				titulo: 'fazer algo3',
				conteudo: 'fazer alguma coisa3',
				status: status.CONCLUIDO,
				adicionadoEm: '28/01',
			},
		]
		const { asFragment } = renderWithRedux(
			<TarefasContainer
				tarefas={tarefas}
				getTarefas={mockGet}
				trocaStatus={mockTrocaStatus}
			/>
		)

		expect(asFragment()).toMatchSnapshot()
	})
	test('deve renderizar 1 tarefa e trocar o status', () => {
		const mockTrocaStatus = jest.fn()
		const mockGet = jest.fn()

		const tarefas = [
			{
				titulo: 'fazer algo1',
				conteudo: 'fazer alguma coisa1',
				status: status.FAZER,
				adicionadoEm: '28/01',
			},
		]
		const { getByText } = renderWithRedux(
			<TarefasContainer
				tarefas={tarefas}
				getTarefas={mockGet}
				trocaStatus={mockTrocaStatus}
			/>
		)

		const tituloT1 = getByText(tarefas[0].titulo)
		const btnFazer = getByText('Fazer')

		expect(tituloT1).toBeDefined()

		fireEvent.click(btnFazer)
		expect(mockTrocaStatus).toHaveBeenCalledTimes(1)
		expect(mockGet).toHaveBeenCalledTimes(1)
	})
})
