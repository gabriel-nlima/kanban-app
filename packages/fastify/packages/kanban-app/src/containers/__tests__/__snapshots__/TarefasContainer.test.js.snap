// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tarefas deve renderizar um quadro de tarefas com 3 tarefas 1`] = `
<DocumentFragment>
  <form
    class="needs-validation"
  >
    <div
      class="form-row"
    >
      <div
        class="col-4"
      >
        <input
          class="form-control mb-2 mr-sm-2"
          data-testid="inputTitulo"
          maxlength="35"
          name="titulo"
          placeholder="Titulo"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="col-5"
      >
        <input
          class="form-control mb-2 mr-sm-2"
          data-testid="inputConteudo"
          maxlength="55"
          name="conteudo"
          placeholder="Descrição"
          type="text"
          value=""
        />
      </div>
      <div
        class="col-3"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          Salvar Tarefa
        </button>
      </div>
    </div>
  </form>
  <div
    class="row text-left"
  >
    <div
      class="col-12 col-md-6 col-lg-6 col-xl-6"
    >
      <h5
        class="text-muted"
      >
        <a
          href="/arquivadas"
        >
          ARQUIVADAS:
        </a>
        <span
          class="badge badge-pill badge-secondary"
          style="margin-left: 6px;"
        >
          0
        </span>
      </h5>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-6 col-md-4 col-lg-4 col-xl-4"
    >
      <h3
        class="text-center text-info"
      >
        A FAZER
        <span
          class="badge badge-pill badge-info"
          style="margin-left: 6px;"
        >
          1
        </span>
      </h3>
      <div
        class="card w-100 h-auto text-white bg-info"
        style="margin-bottom: 5px;"
      >
        <div
          class="card-body text-center"
        >
          <h5
            class="card-title"
          >
            fazer algo1
          </h5>
          <p
            class="card-text text-justify"
          >
            fazer alguma coisa1
          </p>
          <p
            class="card-text text-left "
            style="font-size: 11px;"
          >
            Adicionado em: 28/01
          </p>
          <p
            class="card-text text-left"
            style="font-size: 11px;"
          />
          <a
            class="btn btn-sm btn-light"
            href="/editar"
            style="margin-right: 5px;"
          >
            Editar 
          </a>
          <button
            class="btn btn-sm btn-secondary"
          >
            Fazer
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-6 col-md-4 col-lg-4 col-xl-4"
    >
      <h3
        class="text-center text-warning"
      >
        FAZENDO
        <span
          class="badge badge-pill badge-info"
          style="margin-left: 6px;"
        >
          1
        </span>
      </h3>
      <div
        class="card w-100 h-auto bg-warning"
        style="margin-bottom: 5px;"
      >
        <div
          class="card-body text-center"
        >
          <h5
            class="card-title"
          >
            fazer algo2
          </h5>
          <p
            class="card-text text-justify"
          >
            fazer alguma coisa2
          </p>
          <p
            class="card-text text-left "
            style="font-size: 11px;"
          >
            Adicionado em: 28/01
          </p>
          <p
            class="card-text text-left"
            style="font-size: 11px;"
          />
          <a
            class="btn btn-sm btn-light"
            href="/editar"
            style="margin-right: 5px;"
          >
            Editar 
          </a>
          <button
            class="btn btn-sm btn-secondary"
          >
            Concluir
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-4 col-lg-4 col-xl-4"
    >
      <h3
        class="text-center text-success"
      >
        FEITO
        <span
          class="badge badge-pill badge-info"
          style="margin-left: 6px;"
        >
          1
        </span>
      </h3>
      <div
        class="card w-100 h-auto bg-success"
        style="margin-bottom: 5px;"
      >
        <div
          class="card-body text-center"
        >
          <h5
            class="card-title"
          >
            fazer algo3
          </h5>
          <p
            class="card-text text-justify"
          >
            fazer alguma coisa3
          </p>
          <p
            class="card-text text-left "
            style="font-size: 11px;"
          >
            Adicionado em: 28/01
          </p>
          <p
            class="card-text text-left"
            style="font-size: 11px;"
          />
          <a
            class="btn btn-sm btn-light"
            href="/editar"
            style="margin-right: 5px;"
          >
            Editar 
          </a>
          <button
            class="btn btn-sm btn-secondary"
          >
            Arquivar
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
